web/src/admin/roles/RelatedRoleList.ts

```diff
@@ -96,19 +96,95 @@ export class RelatedRoleList extends Table<Role> {
+
+        // Handle group filtering
+        if (this.targetGroup) {
+            // Always fetch both direct and inherited roles
+            const [directResponse, inheritedResponse] = await Promise.all([
+                new RbacApi(DEFAULT_CONFIG).rbacRolesList({
+                    ...config,
+                    akGroups: [this.targetGroup.pk],
+                }),
+                new RbacApi(DEFAULT_CONFIG).rbacRolesList({
+                    ...config,
+                    inheritedGroupRoles: this.targetGroup.pk,
+                }),
+            ]);
+
+            if (this.showInherited) {
+                // Combine, deduplicate by pk, and sort alphabetically
+                const allRoles = [...directResponse.results, ...inheritedResponse.results];
+                const uniqueRoles = allRoles
+                    .filter(
+                        (role, index, self) => self.findIndex((r) => r.pk === role.pk) === index,
+                    )
+                    .sort((a, b) => a.name.localeCompare(b.name));
+                return {
+                    pagination: directResponse.pagination,
+                    results: uniqueRoles,
+                };
+            }
+            return directResponse;
+        }
+
+        // Handle user filtering - always fetch both direct and inherited roles
+        const [directResponse, inheritedResponse] = await Promise.all([
+            new RbacApi(DEFAULT_CONFIG).rbacRolesList({
+                ...config,
+                users: this.targetUser?.pk ? [this.targetUser.pk] : undefined,
+            }),
+            new RbacApi(DEFAULT_CONFIG).rbacRolesList({
+                ...config,
+                inheritedUserRoles: this.targetUser?.pk,
+            }),
+        ]);
+
+        if (this.showInherited) {
+            // Combine, deduplicate by pk, and sort alphabetically
+            const allRoles = [...directResponse.results, ...inheritedResponse.results];
+            const uniqueRoles = allRoles
+                .filter((role, index, self) => self.findIndex((r) => r.pk === role.pk) === index)
+                .sort((a, b) => a.name.localeCompare(b.name));
+            return {
+                pagination: directResponse.pagination,
+                results: uniqueRoles,
+            };
+        }
+        return directResponse;
     }
 
-    protected columns: TableColumn[] = [
-        [msg("Name"), "name"],
-        [msg("Actions"), null, msg("Row Actions")],
-    ];
+    get columns(): TableColumn[] {
+        // Hide actions column in showInherited mode (view-only)
+        if (this.showInherited) {
+            return [[msg("Name"), "name"]];
+        }
+        return [
+            [msg("Name"), "name"],
+            [msg("Actions"), null, msg("Row Actions")],
+        ];
+    }
 
     renderToolbarSelected(): TemplateResult {
+        // Don't render Remove button in showInherited mode (view-only)
+        if (this.showInherited) {
+            return html``;
```

Comment:
Update function return signature to `SlottedTemplateResult`

```suggestion
            return nothing;
```